srtcore/core.cpp

```diff
@@ -3908,6 +3908,14 @@ EConnectStatus srt::CUDT::craftKmResponse(uint32_t* aw_kmdata, size_t& w_kmdatas
     int hs_flags = SrtHSRequest::SRT_HSTYPE_HSFLAGS::unwrap(m_ConnRes.m_iType);
     if (IsSet(hs_flags, CHandShake::HS_EXT_KMREQ))
     {
+        if (!m_pCryptoControl)
```

Comment:
This lacks an explanation as to why this fixes the problem. This function should at best have:

* The declarations of all mutexes that are locked before calling this function and made sure that the list is stable against all places from which this function is called
* In all places that can modify `m_pCryptoControl` make sure that locking on that place can exclude the possibility of modifying this field anywhere up to the end of this function (mention these places here)